### PR TITLE
Update billing info link in the account page still redirect to stripe…

### LINKED ISSUE #2793: Update billing info link in the account page still redirect to stripe, no matter the payment gateway selected.
If the Update billing flow setting is set to 'portal' in the Stripe gateway setting and for some reason the payment gateway is changed, the Update billing info link in the account page still redirect to stripe, no matter the payment gateway selected.

**To Reproduce**
1. Go to Payment Gateway settings.
2. Select stripe as preferred gateway.
3. Select update billing in customer portal
4. Save settings.
5. Select some other payment gateway.
6. Save settings.
7. Go to any user account page
8. Click update billing info link.  

**Screenshots**
If applicable, please attach a screenshot to make your issue clearer.

**Expected behavior**
 click update billing info shouldn't redirect to stripe customer portal if other gateway is selected.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only Paid Memberships Pro plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Memberlite](https://wordpress.org/themes/memberlite/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
Please share non-sensitive information about your hosting environment such as WordPress version, PHP version, Paid Memberships Pro and any related plugins versions.
```

WP 6.4.2
PHP 8.1.23
</details>

### PR DESCRIPTION
**Pull request by MaximilianoRicoTabo**
Bail if selected gateway isn't Stripe.

### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Resolves #2793 

### How to test the changes in this Pull Request:
Follow steps described in the issue above

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by dparker1005 — COMMENTED** [review]
I like the refactor to return early when 'portal' is not the selected update billing flow.

As for making sure redirects only happen when Stripe is the selected gateway, the proposed code will work, but we can acutually handle this when hooking the `pmpro_billing_preheader_stripe_customer_portal()` method here: https://github.com/strangerstudios/paid-memberships-pro/blob/e5366d0a1a4cddea5f717991eac176cf445abb2d/classes/gateways/class.pmprogateway_stripe.php#L174

If we move that hook declaration to inside the `$default_gateway == "stripe" || $current_gateway == "stripe"` conditional, we should still prevent redirecting when Stripe isn't the selected gateway, and this will work more similarly to how the rest of the class handles this kind of behavior.

If we want to go a step further, though, subscriptions table makes it easy to check if a subscription is for a particular gateway. With that in mind, I think the ideal solution is to:
1. Get the global `$pmpro_billing_order` that is set in the PMPro Billing preheader
2. Get the subscription for that order
3. If the subscription is not a Stripe subscription, bail

File: classes/gateways/class.pmprogateway_stripe.php [hunk 1]
```diff
@@ -1692,36 +1692,45 @@ public static function pmpro_billing_preheader_stripe_checkout() {
...
-			if ( empty( $error ) ) {
-				// Get the Stripe Customer.
-				$stripe = new PMProGateway_stripe();
-				$customer = $stripe->get_customer_for_user( $user->ID );
-				if ( empty( $customer->id ) ) {
-					$error = __( 'Could not get Stripe customer for user.', 'paid-memberships-pro' );
-				}
-			}
+		//Bail if the order's gateway isn't Stripe
+		if ( 'stripe' !== $pmpo_billing_order->gateway  ) {
```
Comment (dparker1005) [inline classes/gateways/class.pmprogateway_stripe.php:1702]:
```suggestion
		if ( empty( $pmpro_billing_order->gateway ) || 'stripe' !== $pmpro_billing_order->gateway  ) {
```
Fixing variable name typo and making sure that we have a billing order with a gateway set.

Comment (MaximilianoRicoTabo) [reply classes/gateways/class.pmprogateway_stripe.php:1702]: good catch on the typo thanks. What's weird is i'm a hundred percent i debugged this one and i saw the global object containing the gateway 🤔  not sure when i introduced the typo bug, perhaps in the second commit. Anyway thanks for catching it.

Comment (MaximilianoRicoTabo) [reply classes/gateways/class.pmprogateway_stripe.php]: done.

**Review by dparker1005 — APPROVED** [review]

**Pull request merged**